### DISCUSSION TITLE
Generate mex file for .c code from casadi+fatrop

### DISCUSSION THREAD

**Discussion by maskros-m** [casadi/casadi#3873] [Q&A] [answered]
Dear developers and users,

I have been trying to generate a mex file from a generated C code (for use in simulink) following this guideline. 

https://web.casadi.org/blog/s-function/

The C code was sucessfully generated from casadi in python and the library of blasfeo and fatrop were together compiled. However, when running mex in Matlab I encounter the following issue. Those functions appear to be missing in the generated C code.

![Unbenannt](https://github.com/user-attachments/assets/737254d8-c9f6-431f-94c1-94a6f7b4a091)

I tried to search for the definition of these functions in the repo of blasfeo [https://github.com/giaf/blasfeo.git](url), but cannot find their definition. 

Could someone please give me some hints, what the root cause is?

Thanks in advance!

Best regards,
Anh

**Comment by jgillis** [comment]
Is this not resolved by linking in blasfeo.lib in addition to fatrop.lib?

Note that the blasfeo source code may give false impressions on available functions, since a lot of macros are used.

**Reply by maskros-m** [reply]
Hi Joris,

Do you mean linking the library of fatrop and blasfeo in the mex function? If so, their libraries are linked to

```matlab
fold1 = fullfile('C:\Users\ngocngu4\Documents\GitHub\mi_ptceu_pcpc\.venv\Lib\site-packages\casadi\include');
fold2 = fullfile('C:\Users\ngocngu4\Documents\GitHub\mi_ptceu_pcpc\.venv\Lib\site-packages\casadi\lib');
fold3 = fullfile('C:\Users\ngocngu4\Documents\GitHub\mi_ptceu_pcpc\matlab\mdl\Cmpnts\MesoPlanner\eopt_standalone\blasfeo-master');
path = {['-I',fold1],['-I',fold2],['-I',fold3]};

mex(path{:}, 'EOPT_fatrop_call.c', 'EOPT_fatrop.c')
```

**Reply by jgillis** [reply, marked as answer]
Try adding '-lblasfeo', '-lfatrop' and '-LC:\Users\ngocngu4\Documents\GitHub\mi_ptceu_pcpc.venv\Lib\site-packages\casadi\lib'

**Reply by maskros-m** [reply]
Thank you very much Joris! it is working now!

**Reply by maskros-m** [reply]
Hi Joris, sorry for further disturbance, that might not relate to casadi itself.
I'm trying to configure a second S-function, the same way, it was done in this example. But I don't see all the ports. May I ask whether you needed some further configuration to show all the ports in this example?
[https://web.casadi.org/blog/s-function/](url)
![Unbenannt](https://github.com/user-attachments/assets/d2ed0e2e-ddab-478f-a5bb-32a321d99272)

Thank you in advance!
Best regards,
Anh

**Reply by jgillis** [reply]
One thing to check is making sure that the casadi-exported Functions have unique names (tanslates to symbol namespace).
Perhaps you need to hit the run button for the ports to show up?

**Reply by maskros-m** [reply]
It was my bad not to run it! Thank you very much for your help!
